war/src/main/js/add-item.js

```diff
@@ -0,0 +1,386 @@
+// Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
+var $ = require('jquery-detached').getJQuery();
+
+var getItems = function(root){
+  var d = $.Deferred();
+  $.get(root+'/categories?depth=3').done(
+      function(data){
+        d.resolve(data);
+      }
+  );
+  return d.promise();
+}; 
+
+var jRoot = $('head').attr('data-rooturl');
+
+$.when(getItems(jRoot)).done(function(data){
```

Comment:
@jglick @recena hu? Why would the categories be specific to folders? I guess there is a security context on the plugins?